spec/requests/api/v1/communications/show_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Communications show', type: :request do
+  let!(:user) { create(:user) }
+  let!(:auth_headers) { user.create_new_auth_token }
+  let!(:communication) { create(:communication, published: true) }
+  let!(:unpublished_communication) { create(:communication, published: false) }
+
+  describe 'GET api/v1/communications/<id>' do
+    context 'with authentication' do
+      it 'shows the communication if published' do
+        get "/api/v1/communications/#{communication.id}", headers: auth_headers
+        expect(response).to have_http_status 200
+        response_body = Oj.load(response.body)['communication']
+        expect(response_body['id']).to eq(communication.id)
+        expect(response_body['title']).to eq(communication.title)
+        expect(response_body['text']).to eq(communication.text)
```

Comment:
porqué no hacer `expect(response_body['communication']).to eq(communication.as_json)` como en el otro test? as_json tiene mas cosas que no van acá? again falta la fecha de publicación

Reply:
as_json te trae todo el objeto, con campos como published y cosas así que a priori no irían en la app

Reply:
con esto solucionas todo en una linea
expect(response_body['communication']).to eq(communication.as_json(only: %i[id title text]))